File: src/main/java/com/hoi4utils/clauzewitz/map/province/VictoryPoint.java
package main.java.com.hoi4utils.clauzewitz.map.province;

import main.java.com.hoi4utils.clauzewitz.localization.Localizable;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;
import java.util.Map;

/**
 * localizable: victory point name
 */
public class VictoryPoint implements Localizable {
	static List<VictoryPoint> victoryPoints = new ArrayList<>();
	int value;
	Province province;

	protected VictoryPoint(int province, int value) {
		this.province = Province.of(province);
		this.value = value;
	}

	public static VictoryPoint of(int province, int value) {
		for (VictoryPoint vp : victoryPoints) {
			if (vp.province.id == province) {
				return vp;
			}
		}
		return new VictoryPoint(province, value);
	}

	public int value() {
		return value;
	}

	public void setValue(int value) {
		this.value = value;
	}

	@Override
	public @NotNull Map<Property, String> getLocalizableProperties() {
		return Map.of(Property.NAME, "VP_NAME");
	}

	@Override
	public @NotNull Collection<? extends Localizable> getLocalizableGroup() {
		return victoryPoints;
	}
}

File: src/main/scala/com/hoi4utils/hoi4mod/map/province/VictoryPoint.scala
package com.hoi4utils.hoi4mod.map.province

import com.hoi4utils.hoi4mod.localization.{Localizable, Property}

import scala.collection.mutable



//
/**
 * localizable: victory point name
 *
 * TODO: Remove?
 */
class VictoryPoint protected(province: Int, var value: Int) extends Localizable {
  /**
   * Default method to get the localizable property identifiers and keys.
   *
   * @return a map of localizable property identifiers and keys.
   */
  override def getLocalizableProperties: mutable.Map[Property, String] = ???

  /**
   * Gets the localizable group of objects that this object is a part of.
   *
   * @return the localizable group of objects.
   */
  override def getLocalizableGroup: Iterable[? <: Localizable] = ???
//  this.province = Province.apply(province)
//  private[province] var province: Province = null
//
//  def value: Int = value
//
//  def setValue(value: Int): Unit = {
//    this.value = value
//  }
//
//  @NotNull override def getLocalizableProperties: Map[Property, String] = CollectionConverters.asScala(util.Map.of(Property.NAME, "VP_NAME"))
//
//  @NotNull override def getLocalizableGroup: Iterable[_ <: Localizable] = CollectionConverters.asScala(VictoryPoint.victoryPoints)
//}
//
//object VictoryPoint {
//  private[province] val victoryPoints = new util.ArrayList[VictoryPoint]
//
//  def of(province: Int, value: Int): VictoryPoint = {
//    import scala.collection.JavaConversions._
//    for (vp <- victoryPoints) {
//      if (vp.province.id == province) return vp
//    }
//    new VictoryPoint(province, value)
//  }
}
